### ISSUE TITLE
hdevtools check not working

### ISSUE THREAD

**Issue by schoettl** [hdevtools/hdevtools#3]
My current version of hdevtools does not work anymore:
```
$ hdevtools check ~/.xmonad/xmonad.hs

```
hangs or waits for input and can only be terminated with Ctrl-C (not with Ctrl-D).

This also breaks my Vim/Syntastic. Any ideas?

```
$ hdevtools --version
hdevtools: version 0.1.2.1 (ghc-7.10.2-x86_64-linux, cabal-1.22.4.0)
```

**Comment by ch1bo** [comment]
Hey, for me it just works fine (by printing Xmonad libraries not found :). Is there a `.hdevtools.sock` file after you tried executing / killing `hdevtools`? Furthermore, can you try again in nodaemon mode as follows:

Make sure to have no `.hdevtools.sock` file in your current working directory (or `hdevtools --stop-server`) and execute:
```
hdevtools --start-server --nodaemon
```
and in a second terminal (from the same directory): 
```
hdevtools check ~/.xmonad/xmonad.hs
```
and please post the stdout of both processes.

**Comment by schoettl** [comment]
Thank you for reply! OK, I just executed the hdevtools check command again. After killing it there is no `.hevtools.sock` file in the `pwd`.

```
$ hdevtools --start-server --nodaemon
hdevtools: /usr/lib/ghc-7.10.2/settings: openFile: does not exist (No such file or directory)
$ 
```

The check command still hangs and must be terminated with Ctrl-C
```
$ hdevtools check ~/.xmonad/xmonad.hs

```

What would you suggest? Other ideas than reinstalling GHC?

**Comment by ch1bo** [comment]
Before modifying your system-wide instance I would first try one installed via [stack](https://github.com/commercialhaskell/stack), i.e. by invoking `stack setup` which installs the ghc of the latest lts-haskell version (also ghc-7.10.2 for lts-3.18 I believe) into your home directory (`.stack/tools/...`).

How did you install the system-wide ghc?

**Comment by schoettl** [comment]
I am on Arch Linux and installed GHC via `pacman`. I also use cabal and installed most packages with `cabal install package`.

I did `stack setup`. But `hdevtools` still want to use the system-wide GHC:
```
> hdevtools --start-server --nodaemon
hdevtools: /usr/lib/ghc-7.10.2/settings: openFile: does not exist (No such file or directory)
```

**Comment by ch1bo** [comment]
`hdevtools` used the ghc which is available on its `PATH`, so try changing the environment variable to point also to the stack ghc binary. You can double check this using `which ghc` and also post `ghc --version`.

I am also on arch and on the most recent version of the ghc package, which is the 7.10.3 compiler. At first I expected an incompatibility between hdevtools built against 7.10.2 and then run using the system-wide 7.10.3. But even in this scenario I couldn't reproduce your error.

**Comment by ch1bo** [comment]
Wait a second.. it can't access settings in `/usr/lib/ghc-7.10.2`. Does that exist in your system (as the most recent arch binary is 7.10.3). Anyway an update would be worth a try I guess.

**Comment by schoettl** [comment]
I've updated Arch every day. My GHC is 7.10.3. I guess it's pretty new.

I just did `cabal update && cabal install hdevtools` and now it works!

```
$ ghc --version
The Glorious Glasgow Haskell Compilation System, version 7.10.3
$ hdevtools --version
hdevtools: version 0.1.2.1 (ghc-7.10.3-x86_64-linux, cabal-1.22.5.0)
$ echo /usr/lib/ghc-*
/usr/lib/ghc-7.10.3
```

I did the same update a few days ago before I posted this issue. So but now it seems fixed.

Thank you!

PS: Should I better use `stack` instead of `cabal` for my user-specific Haskell packages?

**Comment by ch1bo** [comment]
Regarding your PS: 
For installing into your home, stack is not much different than cabal-install (the hackage package containing `cabal`), except that stack can cope better with changing versions of libraries and compilers (it maintains them as well in your home). 

When it comes to haskell development, stack is IMHO superior to cabal-install as it builds in a sandbox by default, source dependencies can be specified easier, dependencies can be rebuilt with special flags for your project and more.

TLDR: It allows for faster, easier configurable and better reproducible builds.

**Comment by schoettl** [comment]
Thanks for the advice!
